src/ExportableMesh.cpp

```diff
@@ -121,16 +121,32 @@ ExportableMesh::ExportableMesh(
 
                 ++vertexBufferIndex;
             }
-
+            GLTF::Object* morphTargetNames = new GLTF::MorphTargetNames;
+            size_t morph_count = 0;
             for (auto&& shape : mayaMesh->allShapes())
             {
                 if (shape->shapeIndex.isBlendShapeIndex())
                 {
                     m_weightPlugs.emplace_back(shape->weightPlug);
                     m_initialWeights.emplace_back(shape->initialWeight);
                     glMesh.weights.emplace_back(shape->initialWeight);
+                    MStringArray weightArrays;
+                    MString weight = shape->weightPlug.name();
+                    weight.split('.', weightArrays);
+                    if (weightArrays.length() <= 1) {
+                        static_cast<GLTF::MorphTargetNames*>(morphTargetNames)->addName("morph_" + morph_count);
+                        morph_count++;

```

Comment:
What is `shape->weightPlug.name()` returning? Just the name of attribute? If so, why split on '.'? If it includes the name of the object, followed by a dot, then okay.

I would add a `size()` method to your `GLTF::MorphTargetNames` class, then you don't need `morph_count` here.

This code can be compactly written as:

```
m_morphTargetNames->addName(weightArrays.length() <= 1 
     ? std::string("morph_") + (m_morphTargetNames->size() + 1)
     : std::string(weightArrays[1].asChar())
```
I would certainly remove the `continue` statement.

Reply:
http://help.autodesk.com/view/MAYAUL/2016/ENU/?guid=__cpp_ref_class_m_plug_html

Same thing in 2016.

Reply:
Do you know why you're adding a + to a size? 

> m_morphTargetNames->size() + 1

Sizes start from 1.

Reply:
> Do you know why you're adding a + to a size?
> 
> > m_morphTargetNames->size() + 1
> 
> Sizes start from 1.

If nothing is added to `m_morphTargetNames`, `size()` should return `0`. Of course you first need to add a `size()` method to the `GTLF::MorphTargetNames`. It's actually okay for me to just expose the `std::vector <std::string> names` in `GTLF::MorphTargetNames`, the whole GLTF lib has a rather non-encapsulation style. Then you can remove `addName` etc, and just work on the `vector` directly.

Reply:
Hm, I've already submitted the size getter, should I change it?

Reply:
Not really, auto-generated indices can start at zero, that's fine for me. The code shouldn't be triggered anyway, since the plug name should always be set.